tests/TestCase.php

```diff
@@ -3,8 +3,34 @@
 namespace Tests;
 
 use Illuminate\Foundation\Testing\TestCase as BaseTestCase;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\Relations\BelongsToMany;
 
 abstract class TestCase extends BaseTestCase
 {
     use CreatesApplication;
+
+    protected function testHasManyRelation($related, $foreignKey, $relation)
+    {
+        $this->assertInstanceOf(HasMany::class, $relation);
+        $this->assertInstanceOf($related, $relation->getRelated());
+        $this->assertEquals($foreignKey, $relation->getForeignKeyName());
+    }
+
+    protected function testBelongsToRelation($related, $foreignKey, $ownerKey, $relation)
+    {
+        $this->assertInstanceOf(BelongsTo::class, $relation);
+        $this->assertInstanceOf($related, $relation->getRelated());
+        $this->assertEquals($ownerKey, $relation->getOwnerKeyName());
+        $this->assertEquals($foreignKey, $relation->getForeignKeyName());
+    }
+
+    protected function testBelongsToManyRelation($relation, $foreignPivotKey, $relatedPivotKey, $related)
+    {
+        $this->assertInstanceOf(BelongsToMany::class, $relation);
+        $this->assertInstanceOf($related, $relation->getRelated());
+        $this->assertEquals($foreignPivotKey, $relation->getForeignPivotKeyName());
+        $this->assertEquals($relatedPivotKey, $relation->getRelatedPivotKeyName());
+    }
```

Comment:
giải thích các function này

Reply:
$this->assertInstanceOf(HasMany::class, $relation);  // kiểm tra xem quan hệ (biến $relation) có phải là kiểu HasMany hay không (tương tự với các kiểu quan hệ khác)
$this->assertInstanceOf($related, $relation->getRelated());  // kiểm tra xem model (biến $related) có đúng với quan hệ không
$this->assertEquals($foreignKey, $relation->getForeignKeyName());  // kiểm tra xem có đúng khoá ngoại không
$this->assertEquals($ownerKey, $relation->getOwnerKeyName());  // kiểm tra xem khoá chính có đúng ko
$this->assertEquals($foreignPivotKey, $relation->getForeignPivotKeyName());  // kiểm tra xem khoá ngoại ở bảng trung gian có đúng ko
$this->assertEquals($relatedPivotKey, $relation->getRelatedPivotKeyName());  // kiểm tra xem khoá liên quan (khoá liên quan là khoá ngoại thứ 2 ở bảng trung gian) có đúng không